### DISCUSSION TITLE
update the lr by iteration in epochbasedrunner

### DISCUSSION THREAD

**Discussion by Jacobew** [open-mmlab/mmdetection#8956] [Q&A]
Hi, how can I update the learning rate by iteration but not by epoch when using an epochbasedrunner?

My other question is: which part of the codes should I modify to substitute epochbasedrunner with iterbasedrunner?

Looking forward for the reply :)

**Comment by zachcoleman** [comment]
Set the `by_epoch` argument in the LR config to `False` for iter-based updates like:
```python
lr_config = dict(
    policy="step",
    warmup="linear",
    warmup_iters=500,
    warmup_ratio=0.001,
    step=[70_000],
    by_epoch=False,
)
```

**Comment by zachcoleman** [comment]
In the config you can also specify the runner type like:
```python
runner = dict(type="IterBasedRunner", max_iters=200_000)
```
